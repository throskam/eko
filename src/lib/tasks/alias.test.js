jest.mock('../cio')
jest.mock('../config')

const cio = require('../cio')
const config = require('../config')

// Keep real implementation of message.
cio.message = jest.requireActual('../cio').message

const alias = require('./alias')

beforeEach(() => {
  jest.resetAllMocks()
})

it('should list aliases', async () => {
  expect.assertions(1)

  const messages = []

  const aliases = [{
    name: 'my-first-alias',
    command: 'my-first-command --with-params'
  }, {
    name: 'my-second-alias',
    command: 'my-second-command'
  }]

  config.aliases.list.mockResolvedValue(aliases)
  cio.log.mockImplementation(message => messages.push(message))

  await alias()

  expect(messages).toMatchSnapshot()
})

it('should add an alias', async () => {
  expect.assertions(1)

  const name = 'my-alias'
  const command = 'my-command'
  const option = {
    force: true
  }

  await alias(name, command, option)

  expect(config.aliases.add).toHaveBeenCalledWith({ name, command }, option.force)
})

it('should remove an alias', async () => {
  expect.assertions(1)

  const name = 'my-alias'
  const option = {
    delete: true
  }

  await alias(name, '', option)

  expect(config.aliases.remove).toHaveBeenCalledWith(name)
})

it('should return when there is no command', async () => {
  expect.assertions(1)

  const name = 'my-alias'

  await alias(name)

  expect(config.aliases.add).not.toHaveBeenCalled()
})
