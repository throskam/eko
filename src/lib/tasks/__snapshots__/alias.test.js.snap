// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should list aliases 1`] = `
Array [
  "[36m‚óè [39mmy-first-alias [36m $ [39mmy-first-command --with-params",
  "[36m‚óè [39mmy-second-alias[36m $ [39mmy-second-command",
]
`;
